sensor_msgs/test/test_pointcloud_iterator.cpp

```diff
@@ -124,3 +124,51 @@ TEST(sensor_msgs, PointCloud2Iterator)
   }
   EXPECT_EQ(i, n_points);
 }
+
+TEST(sensor_msgs, PointCloud2Resize)
+{
+  // Check if size will retain when width and height given explicitly (width == 1)
+  size_t n_points = 4;
+  sensor_msgs::msg::PointCloud2 cloud_msg_1;
+  cloud_msg_1.height = static_cast<uint32_t>(n_points);
+  cloud_msg_1.width = 1;
+  sensor_msgs::PointCloud2Modifier modifier(cloud_msg_1);
+  modifier.setPointCloud2FieldsByString(2, "xyz", "rgb");
+  modifier.resize(1 * n_points);
+  EXPECT_EQ((unsigned int)1, cloud_msg_1.width);
```

Comment:
Prefer using `static_cast` and `uint32_t` (which is the actual type of the width and height members).

It might also be fine to just declare the literals as unsigned, like `1ul`.